server/data/data-jmap/src/test/java/org/apache/james/jmap/api/vacation/AbstractVacationRepositoryTest.java

```diff
@@ -59,18 +60,302 @@ public void retrieveVacationShouldReturnDefaultValueByDefault() {
+            .fromDate(MissableOptional.ofOptional(VACATION.getFromDate()))
+            .toDate(MissableOptional.ofOptional(VACATION.getToDate()))
+            .isEnabled(MissableOptional.of(VACATION.isEnabled()))
+            .build();
 
-        assertThat(vacationRepository.retrieveVacation(ACCOUNT_ID).join()).isEqualTo(VACATION_1);
+        vacationRepository.modifyVacation(ACCOUNT_ID, vacationUpdate).join();
+
+        assertThat(vacationRepository.retrieveVacation(ACCOUNT_ID).join())
+            .isEqualTo(VACATION);
+    }
+
+    @Test
+    public void modifyVacationShouldAllowEmptyUpdates() {
+        VacationUpdate vacationUpdate = VacationUpdate.builder()
+            .build();
+
+        vacationRepository.modifyVacation(ACCOUNT_ID, vacationUpdate).join();
+
+        assertThat(vacationRepository.retrieveVacation(ACCOUNT_ID).join())
+            .isEqualTo(VacationRepository.DEFAULT_VACATION);
+    }
+
+    @Test
+    public void emptyUpdatesShouldNotChangeExistingVacations() {
+        // Given
+        vacationRepository.modifyVacation(ACCOUNT_ID,
+            VacationUpdate.builder()
+                .subject(MissableOptional.ofOptional(VACATION.getSubject()))
+                .textBody(MissableOptional.ofOptional(VACATION.getTextBody()))
+                .htmlBody(MissableOptional.ofOptional(VACATION.getHtmlBody()))
+                .fromDate(MissableOptional.ofOptional(VACATION.getFromDate()))
+                .toDate(MissableOptional.ofOptional(VACATION.getToDate()))
+                .isEnabled(MissableOptional.of(VACATION.isEnabled()))
+                .build())
+            .join();
+
+        // When
+        vacationRepository.modifyVacation(ACCOUNT_ID, VacationUpdate.builder()
+            .build())
+            .join();
+
+        // Then
+        assertThat(vacationRepository.retrieveVacation(ACCOUNT_ID).join())
+            .isEqualTo(VACATION);
+    }
+
+    @Test
+    public void nullUpdateShouldResetSubject() {
+        // Given
+        vacationRepository.modifyVacation(ACCOUNT_ID,
+            VacationUpdate.builder()
+                .subject(MissableOptional.ofOptional(VACATION.getSubject()))
+                .textBody(MissableOptional.ofOptional(VACATION.getTextBody()))
+                .htmlBody(MissableOptional.ofOptional(VACATION.getHtmlBody()))
+                .fromDate(MissableOptional.ofOptional(VACATION.getFromDate()))
+                .toDate(MissableOptional.ofOptional(VACATION.getToDate()))
+                .isEnabled(MissableOptional.of(VACATION.isEnabled()))
+                .build())
+            .join();
+
+        // When
+        vacationRepository.modifyVacation(ACCOUNT_ID, VacationUpdate.builder()
+            .subject(MissableOptional.ofNull())
+            .build())
+            .join();
+
+        // Then
+        assertThat(vacationRepository.retrieveVacation(ACCOUNT_ID).join())
+            .isEqualTo(Vacation.builder()
```

Comment:
You are right. It is because I am testing two things : 
 - Old fields were not updated
 - New field is null

I will add an explicit check using extracting for the subject.